### ISSUE TITLE
sensuctl dump does not honor the --namespace flag

### ISSUE THREAD

**Issue by cwjohnston** [sensu/sensu-go#4166]
## Expected Behavior

As a Sensu user I can use `sensuctl dump` with the `--namespace NAMESPACE` flag to export namespaced resources matching only the specified namespace.

## Current Behavior

`sensuctl dump` exports namespaced resources from all namespaces despite use of `--namespace NAMESPACE` flag.

## Steps to Reproduce (for bugs)

1. Create a namespace in addition to `default`, e.g. `test`, 
2. populate the `test` namespace with checks and other namespaced resources
3. Run `sensuctl dump assets,checks,entities,events,filters,handlers,hooks,mutators,roles,rolebindings,silenced --namespace test --format wrapped-json | jq .metadata.namespace`
4. note that both `default` and `test` are present in the output

## Context

In Sensu Go 6.0 and later, namespaces cannot be deleted unless they are empty. If a user wishes to delete a namespace, they must delete all the resources in that namespace. Theoretically this is easy, as we can use `sensuctl dump` to collect all the resources in a file and then delete them with `sensuctl delete -f $FILE`.

Due to the current behavior of `sensuctl dump` a user must iterate over all the namespaced resource types (listed above in steps to reproduce) using commands like `sensuctl $TYPE list --namespace test --format wrapped-json | tee -a test-resources.json` to collect all the resources which reference this namespace.

As `--namespace` is listed as a global flag in the usage for `sensuctl dump` I find it surprising that invoking it does not work as expected.

## Your Environment

* Sensu version used (sensuctl, sensu-backend, and/or sensu-agent): 6.2.1
* Installation method (packages, binaries, docker etc.): packages
* Operating System and version (e.g. Ubuntu 14.04): debian buster

**Comment by palourde** [comment]
@cwjohnston Just to confirm my suspicion, could you run this command?

```
sensuctl dump assets,checks,entities,events,filters,handlers,hooks,mutators,roles,rolebindings,silenced --namespace test --format wrapped-json | jq '.type + " in "+. .metadata.namespace'
```

It should return something like this, which tells us the resource type that was returned and in which namespace:

```
"CheckConfig in test"
"Event in default"
"Event in default"
"Role in test"
"RoleBinding in test"
```

I believe the problem only exists with events (AFAIK, but it might happen with some other resources) because the `URIPath` method of corev2.Event does not build an URL with the namespace within its path in some instances (not sure why though) https://github.com/sensu/sensu-go/blob/20c4be0a485a49454b176a9dd1c7dba5b42d8947/api/core/v2/event.go#L38-L46

**Comment by cwjohnston** [comment]
@palourde you're right, all the resources in "default" namespace are events:

```
$ sensuctl dump assets,checks,entities,events,filters,handlers,hooks,mutators,roles,rolebindings,silenced --namespace test --format wrapped-json | jq '.type + " in "+. .metadata.namespace'
"Asset in test"
"CheckConfig in test"
"Entity in test"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Role in test"
"Role in test"
"RoleBinding in test"
"RoleBinding in test"
```

**Comment by cwjohnston** [comment]
I expected #4167 to be included in 6.2.2 release and fix this issue, but still seeing it in my upgraded environment:

```
[root@backend01 ~]# sensuctl version
sensuctl version 6.2.2+ee, enterprise edition, build f269b86dd1ce5ac9eddc9cec5f3e1cc27b6d5988, built 2021-01-14T17:58:28Z, built with go1.13.15
[root@backend01 ~]# curl -k https://localhost:8080/version
{"etcd":{"etcdserver":"3.3.22","etcdcluster":"3.3.0"},"sensu_backend":"6.2.2"}
[root@backend01 ~]# sensuctl dump assets,checks,entities,events,filters,handlers,hooks,mutators,roles,rolebindings,silenced --namespace test --format wrapped-json | jq '.type + " in "+. .metadata.namespace'
"Asset in test"
"CheckConfig in test"
"Entity in test"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Event in default"
"Role in test"
"Role in test"
"RoleBinding in test"
"RoleBinding in test"
```

**Comment by amdprophet** [comment]
It seems that the fix for this wasn't included in 6.2.2. We just released `v2.6.0` of the `api/core/v2` module which will be included in the upcoming Sensu Go 6.2.3 release.
